static/js/collapsible-filters/interactions.js

```diff
@@ -240,4 +241,55 @@ export default class Interactions {
       yxtFooter.classList.add(this.templateName);
     }
   }
+
+  /**
+   * Use media query with $breakpoint-expanded-filters to determines if the
+   * page is in collapsible filters view.
+   * 
+   * @returns {boolean} whether the viewport is in collapsible filters view
+   */
+  isCollapsibleFiltersView() {
+    const mediaQuery = window.matchMedia(`(min-width: ${ScssVariables.cssBreakpointExpandedFilters})`);
+    return !mediaQuery.matches;
+  }
+
+  /**
+   * Add a media query event listener when there's a width view port change
+   * relative to $breakpoint-expanded-filters.
+   *
+   * @param {Function} handleViewportChange callback function to invoke when there's a media query event
+   */
+  addExpandedFiltersListener(handleViewportChange) {
+    const mediaQuery = window.matchMedia(`(min-width: ${ScssVariables.cssBreakpointExpandedFilters})`);
+    if ('addEventListener' in mediaQuery) {
+      mediaQuery.addEventListener('change', e => handleViewportChange(e.matches));
+     } else if ('addListener' in mediaQuery) {
+      mediaQuery.addListener(e => handleViewportChange(e.matches));
```

Comment:
I guess we don't need to worry about when addListener doesn't exist since it's compatible with basically all browsers?

Reply:
Yea, addListener works for all browser but the more modern ones will use addEventListener (e.g. ie only works for addListener, but chrome have support for addEventListener)